internal/controller/ngrok/kubernetesoperator_controller.go

```diff
@@ -350,12 +351,21 @@ func calculateFeaturesEnabled(ko *ngrokv1alpha1.KubernetesOperator) []string {
 func (r *KubernetesOperatorReconciler) findOrCreateTLSSecret(ctx context.Context, ko *ngrokv1alpha1.KubernetesOperator) (secret *v1.Secret, err error) {
 	secret = &v1.Secret{}
 	err = r.Client.Get(ctx, client.ObjectKey{Namespace: ko.GetNamespace(), Name: ko.Spec.Binding.TlsSecretName}, secret)
-	if err == nil {
+	if !apierrors.IsNotFound(err) {
 		return
 	}
 
-	if !apierrors.IsNotFound(err) {
-		return
+	if err == nil {
+		isValid := secret.Type == v1.SecretTypeTLS &&
+			secret.Data["tls.key"] != nil &&
+			secret.Data["tls.crt"] != nil &&
```

Comment:
I think we can remove the `secret.Data["tls.crt"]` check. I think this may cause the secret's key and CSR to change more frequently than needed. The `findOrCreateTLSSecret` is only responsible for creating the initial key and CSR. The `updateTLSSecretCert` will add the cert to the secret.